From c5929f3d218249b498d0ca215f38d794f25351cf Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Tue, 5 Sep 2017 17:58:02 +0800
Subject: [PATCH 2517/5242] MLK-16374-2: PxP: add new format support for as
 and out buffer

commit  9b314cfc70579134b9fb90efe725c0757f1b45bc from
https://source.codeaurora.org/external/imx/linux-imx.git

1) add PXP_PIX_FMT_BGRA32 format support for AS buffer
2) add PXP_PIX_FMT_BGRA32 format support for OUT buffer

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Reviewed-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 4eeefd54cdbb8d1d3ffd64bd444ea7b759101a35)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/pxp/pxp_dma_v3.c |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/dma/pxp/pxp_dma_v3.c b/drivers/dma/pxp/pxp_dma_v3.c
index 22be1bc..633bcd9 100644
--- a/drivers/dma/pxp/pxp_dma_v3.c
+++ b/drivers/dma/pxp/pxp_dma_v3.c
@@ -1291,6 +1291,7 @@ static uint32_t pxp_parse_as_fmt(uint32_t format)
 	uint32_t fmt_ctrl;
 
 	switch (format) {
+	case PXP_PIX_FMT_BGRA32:
 	case PXP_PIX_FMT_ARGB32:
 		fmt_ctrl = BV_PXP_AS_CTRL_FORMAT__ARGB8888;
 		break;
@@ -1335,6 +1336,7 @@ static uint32_t pxp_parse_out_fmt(uint32_t format)
 	uint32_t fmt_ctrl;
 
 	switch (format) {
+	case PXP_PIX_FMT_BGRA32:
 	case PXP_PIX_FMT_ARGB32:
 		fmt_ctrl = BV_PXP_OUT_CTRL_FORMAT__ARGB8888;
 		break;
@@ -1892,6 +1894,7 @@ static bool fmt_as_support(uint32_t format)
 	case PXP_PIX_FMT_ARGB32:
 	case PXP_PIX_FMT_RGBA32:
 	case PXP_PIX_FMT_XRGB32:
+	case PXP_PIX_FMT_BGRA32:
 	case PXP_PIX_FMT_ARGB555:
 	case PXP_PIX_FMT_ARGB444:
 	case PXP_PIX_FMT_RGBA555:
@@ -1910,6 +1913,7 @@ static bool fmt_out_support(uint32_t format)
 	switch (format) {
 	case PXP_PIX_FMT_ARGB32:
 	case PXP_PIX_FMT_XRGB32:
+	case PXP_PIX_FMT_BGRA32:
 	case PXP_PIX_FMT_RGB24:
 	case PXP_PIX_FMT_ARGB555:
 	case PXP_PIX_FMT_ARGB444:
-- 
1.7.9.5

